cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_check_manager.py

```diff
@@ -0,0 +1,375 @@
+
+
+class HealthCheckConfigLoader:
+    """
+    Health Check configuration loader.
+
+    Loads Health Check configuration from cluster configuration file according to the node type,
+    queue and compute resource.
+    """
+
+    @staticmethod
+    def _load_cluster_config(input_file_path: str) -> str:
+        """Load cluster config file."""
+        return yaml.load(StringIO(_read_file(input_file_path)), Loader=yaml.SafeLoader)
+
+    def load_configuration(
+        self, health_check_manager_config: HealthCheckManagerConfig, args: List[str]
+    ) -> HealthCheckConfig:
+        """
+        Load Health Check configuration.
+
+        :param health_check_manager_config: Health Check Manager configuration
+        :param args: command line arguments
+        :return: Health Check configuration object
+        """
+        cluster_config = self._load_cluster_config(args.cluster_configuration)
+
+        log.debug("Cluster config: %s", cluster_config)
+
+        health_checks = []
+        try:
+            if args.node_type == "ComputeFleet":
+                queues = cluster_config["Scheduling"]["SlurmQueues"]
+
+                queue = next((queue for queue in queues if queue["Name"] == args.queue_name), None)
+                compute_resource = next(
+                    (
+                        compute_resource
+                        for compute_resource in queue["ComputeResources"]
+                        if compute_resource["Name"] == args.compute_resource_name
+                    ),
+                    None,
+                )
+
+                for health_check_key, health_check_value in compute_resource["HealthChecks"].items():
+                    if health_check_key == "CustomChecks":
+                        # not yet implemented
+                        pass
+                    else:
+                        health_check = HealthCheckDefinition(
+                            name=health_check_key,
+                            is_managed=True,
+                            is_enabled=bool(
+                                health_check_value.get("Enabled")
+                                if health_check_value.get("Enabled") is not None
+                                else queue["HealthChecks"][health_check_key].get("Enabled", False)
+                            ),
+                            check_path=ManagedHealthCheckName[health_check_key.upper()].get_health_check_path(
+                                health_check_manager_config.managed_health_check_dir
+                            ),
+                        )
+                        health_checks.append(health_check)
+
+        except (KeyError, TypeError) as err:
+            if hasattr(err, "message"):
+                err = err.message
+            log.warning("Ignore missing %s in config", err)
+
+        health_check_conf = HealthCheckConfig(
+            region=args.region,
+            stack_name=args.stack_name,
```

Comment:
Why do you need the stack name? I mean everything is cluster/stack specific, why do we need to use this value?

Reply:
removed, also region, which is not used